components/esp32/cpu_start.c

```diff
@@ -161,6 +170,16 @@ void IRAM_ATTR call_start_cpu0()
 
     ESP_EARLY_LOGI(TAG, "Pro cpu up.");
 
+    /* Initialize heap allocator. WARNING: This *needs* to happen *after* the app cpu has booted.
```

Comment:
I haven't looked closely yet, but I'm a little concerned about the point mentioned in this comment.

Reply:
gee, i should really read comments better, shouldn't i? sorry.
ok, a bit more tap-dancing is required, but it's done now.